tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc

```diff
@@ -6081,13 +6275,49 @@ Status ConvertDepthSpaceShuffle(OpConverterParams* params) {
   }
   if (params->validation_only) return Status::OK();
 
+  // Adjust dimensions for explicit batch mode, by inserting a batch dim and
+  // shifting (and possibly incrementing) the rest of the dims. This can be used
+  // to transform permutation from implicit to explicit batch mode. Examples:
+  // - permutation {2,1,0} -> {0, 3, 2, 1},
+  // - reshape {2,1,0} -> {N, 2, 1, 0}.
+  auto adjust_perm = [](int n, nvinfer1::Permutation perm,
+                        bool use_implicit_batch) {
+    if (use_implicit_batch) return perm;
+    for (int i = n; i > 0; i--) {
+      perm.order[i] = perm.order[i - 1] + 1;
+    }
+    perm.order[0] = 0;
+    return perm;
+  };
+  auto adjust_reshape = [](int N, nvinfer1::Dims dims,
```

Comment:
Let's add  this comment:
Adjust a reshape constructed at implicit batch mode for explicit batch mode. In particular, we need to insert the batch dimension size to the beginning of all the dimension sizes. Example:
reshape {20,10,30} for implicit batch mode becomes reshape {N,20,10,30} for explicit batch mode.